examples/cache.ex

```diff
@@ -0,0 +1,76 @@
+defmodule ExampleSupervisor do
+  def start do
+    import Supervisor.Spec
+
+    # List comprehension creates a consumer per cpu core
+    children = for i <- 1..System.schedulers_online, do: worker(ExampleConsumer, [], id: i)
+
+    Supervisor.start_link(children, strategy: :one_for_one)
+  end
+end
+
+defmodule ExampleConsumer do
+  use Nostrum.Consumer
+
+  alias Nostrum.Api
+
+  require Logger
+
+  def start_link do
+    Consumer.start_link(__MODULE__, :state)
+  end
+
+  def handle_event({:MESSAGE_CREATE, {msg}, ws_state}, state) do
+    # Let's do a little command 'parsing'
+    # We first grab the content, split it into an array and then convert the array to a tuple
+    args = msg.content |> String.split(" ") |> List.to_tuple
+
+    case elem(args, 0) do # We check the very first arg to see the command
+      "!userinfo" -> # !userinfo user_id
+        if tuple_size(args) == 2 do # Check to make sure there are 2 arguments, meaning the command, and then an argument
+
+          # We grab the second thing from the tuple
+          user_id = elem(args, 1)
+
+          if user_id != nil do
+            try do
+              # try and convert the supplied input to an integer
+              user_id = user_id |> String.to_integer
+
+              # First, we try and grab the data from the cache.
+              # If this fails, it will try the next thing in the clause which in this case would be sending out a request to Discord.
+              # If this fails, it will go over into the else statement down below and print out that it can't find anyone/anything.
+              with {:ok, user} <- Nostrum.Cache.UserCache.get(id: user_id),
+                                  Api.get_user(user_id),
+                   {:ok, channel} <- Nostrum.Cache.ChannelCache.get(id: msg.channel_id),
+                                     Api.get_channel(msg.channel_id),
+                   {:ok, guild} <- Nostrum.Cache.Guild.GuildServer.get(id: channel.guild_id),
+                                   Api.get_guild(channel.guild_id)
+
+              do
+                Api.create_message!(msg.channel_id, "#{user_id} belongs to a person named: #{user.username}\nMessage sent in: #{channel.name}\nChannel is in: #{guild.name}")
+              else
+                # For this, we just print out we can't find anyone while using the error from the with statement. If it can't find someone, it will print out `user_not_found` as the reason.
+                {:error, reason} ->
```

Comment:
If we decide to remove the try, this should include a case for failing to parse the integer.